pims/cine.py

```diff
@@ -288,13 +358,52 @@ def __init__(self, filename):
                                                    })
         self.stack_meta_data['trigger_time'] = self.trigger_time
 
+        ### IMAGES
+        # TODO: add support for reading sequence within the same framework, when data
+        # has been saved in another format (.tif, image sequence, etc)
+        if splitext(filename)[-1] in ['.cine', '.cci', '.cin']:
+            self.image_locations = self._unpack('%dQ' % self.image_count,
+                                               self.off_image_offsets)
+            if type(self.image_locations) not in (list, tuple):
+                self.image_locations = [self.image_locations]
+
+    def _clean_setup_dict(self):
+        """Remove obsolete fields and trailing blank characters \x00."""
+        # Filter out 'Res_' (obsolete) fields
+        setup = self.setup_fields_dict
+        k_res = [k for k in setup.keys() if k.startswith('Res_')]
+        for k in k_res:
+            del setup[k]
+        # Remove blank characters
+        self._remove_trailing_x00(setup)
+        # Format f_tone properly
+        tone = setup['f_tone']
+        setup['f_tone'] = tuple((tone[2*k], tone[2*k+1])\
+                                for k in range(setup['tone_points']))
+        return None
+
+    def _remove_trailing_x00(self, dic):
+        for k, v in dic.items():
+            if isinstance(v, bytes):
+                try:
+                    dic[k] = v.decode('utf8').replace('\x00', '')
+                except:
```

Comment:
This block, and the one below, need to catch a specific exception class or list of classes.

Reply:
Done.
I finally put the function in the module (at the bottom of the file), rather than a class method.